### ISSUE TITLE
Question: How to start a HTTPS connection using Slack Bolt Framework + Flask

### ISSUE THREAD

**Issue by Ashok-senthilarasan** [slackapi/bolt-python#765]
(Describe your issue and goal here)
I am trying to run slack app along with flask ssl enabled (self signed certificate).  When I send block actions request I am getting "This app responded with status code 503" . 

referred the following github issue 
https://github.com/slackapi/bolt-python/issues/735

#### The `slack_bolt` version

```
slack-bolt==1.15.3
slack-sdk==3.19.3
```

#### Python runtime version

```
Python 3.9.6
```
Flask version
```
Flask==2.2.2
Flask-RESTful==0.3.9
```

#### OS info
```
ProductName:            macOS
ProductVersion:         13.0
BuildVersion:           22A380
Darwin Kernel Version 22.1.0: Sun Oct  9 20:15:09 PDT 2022; root:xnu-8792.41.9~2/RELEASE_ARM64_T6000
```
#### Steps to reproduce:

app.py  file 
```python
from flask import Flask
from flask_restful import Api
from resources.actions import Actions

app = Flask(__name__, instance_relative_config=True)
api = Api(app)
api.add_resource(
    Actions,
    "/actions",
)

if __name__ == "__main__":
    context = ("local.crt", "local.key")
    app.run(host="0.0.0.0", port=5001, ssl_context=context, debug=True)
```
resources/actions.py file 
```python
from flask import request
from flask_restful import Resource
from slack_bolt import App
from slack_bolt.adapter.flask import SlackRequestHandler
from slack_sdk import WebClient

import json
import os
import logging
import ssl
import redis
import certifi

class Actions(Resource):
    def __init__(self, **kwargs):
        self.c = "hello"

    def post(self):
        return handler.handle(request)

ssl_context = ssl.create_default_context(cafile=certifi.where())
slack_app = App(
    client=WebClient(token=os.environ["SLACK_TOKEN"], ssl=ssl_context),
)
handler = SlackRequestHandler(slack_app)

@slack_app.action("button_click")
def action(body, ack):
    ack()
```
python3 app.py 

### Expected result:

How to run slack with flask  with self signed certificates. 

### Actual result:

When I send block actions request I am getting "This app responded with status code 503" .  I don't see any logs/error message in the application logs for the request. 

If I  remove `ssl_context=context` in the app.run it works fine.  slack app responds after removing ssl context. 
## Requirements

Please read the [Contributing guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

**Comment by mwbrooks** [comment]
Hey @Ashok-senthilarasan 👋🏻 Thanks for reaching out and referencing issue #735!

I wanted to let you know that we've seen your question and a teammate will help you out soon! Since it's Friday and a local holiday for some of the team, we may not respond for a few days. Unfortunately, I'm not experienced enough in self-signed certificates using Flask to lend you helpful advice.

If you happen to find a solution before receiving a response, we'd appreciate a comment here with your solution! 🙇🏻

**Comment by seratch** [comment]
Hi @Ashok-senthilarasan, the `ssl_context` is supposed to be used for the `client `, which handles your outgoing HTTPS requests towards the Slack API servers. Therefore, you cannot use your own cert for it. The reason that your app does not work with it is that all the code that performs Slack API calls such as auth.test in the built-in middleware fails.

If I understand what you want to do here correctly, you're trying to enable your Flask web app to serve HTTPS endpoints with your cert, right? In this case, you don't need to touch the `client` at all.

I hope this helps.

**Comment by Ashok-senthilarasan** [comment]
Hi @mwbrooks, @seratch, Thanks looking into this issue. 

> If I understand what you want to do here correctly, you're trying to enable your Flask web app to serve HTTPS endpoints with your cert, right?

Yes, that's correct. I am trying to enable HTTPS for my Flask we app with my own cert. 

I removed `ssl_context` from client and ran Flask web app with `ssl_context` but it sill failing with same error. 

logs from the test. 
```
INFO:werkzeug:WARNING: This is a development server. Do not use it in a production deployment. Use a production WSGI server instead.
 * Running on all addresses (0.0.0.0)
 * Running on https://127.0.0.1:5001
INFO:werkzeug:Press CTRL+C to quit
INFO:werkzeug: * Restarting with stat
DEBUG:slackapp:Sending a request - url: https://www.slack.com/api/auth.test, query_params: {}, body_params: {}, files: {}, json_body: None, headers: {'Content-Type': 'application/x-www-form-urlencoded', 'Authorization': '(redacted)', 'User-Agent': 'Bolt/1.15.3 Python/3.9.6 slackclient/3.19.3 Darwin/22.1.0'}
DEBUG:slackapp:Received the following response - status: 200, headers: {'date': 'Sat, 12 Nov 2022 17:49:29 GMT', 'server': 'Apache', 'x-powered-by': 'HHVM/4.153.1', 'access-control-allow-origin': '*', 'referrer-policy': 'no-referrer', 'x-slack-backend': 'r', 'x-slack-unique-id': 'Y2_cqU9WdWNPdATXOCpouAAAEBk', 'strict-transport-security': 'max-age=31536000; includeSubDomains; preload', 'access-control-allow-headers': 'slack-route, x-slack-version-ts, x-b3-traceid, x-b3-spanid, x-b3-parentspanid, x-b3-sampled, x-b3-flags', 'access-control-expose-headers': 'x-slack-req-id, retry-after', 'x-oauth-scopes': 'chat:write,files:write,users:read,users:read.email,channels:history,channels:read', 'expires': 'Mon, 26 Jul 1997 05:00:00 GMT', 'cache-control': 'private, no-cache, no-store, must-revalidate', 'pragma': 'no-cache', 'x-xss-protection': '0', 'x-content-type-options': 'nosniff', 'x-slack-req-id': 'beaaf73bbbfff514d7fd14c97bde4cb4', 'vary': 'Accept-Encoding', 'content-type': 'application/json; charset=utf-8', 'x-envoy-upstream-service-time': '110', 'x-backend': 'main_normal main_bedrock_normal_with_overflow main_canary_with_overflow main_bedrock_canary_with_overflow main_control_with_overflow main_bedrock_control_with_overflow', 'x-server': 'slack-www-hhvm-main-iad-ohbx', 'x-slack-shared-secret-outcome': 'no-match', 'via': 'envoy-www-iad-sf8a, envoy-edge-fra-ryae', 'x-edge-backend': 'envoy-www', 'x-slack-edge-shared-secret-outcome': 'no-match', 'connection': 'close', 'transfer-encoding': 'chunked'}, body: {"ok":true,"url":"https:\/\/focahq.slack.com\/","team":"foca","user":"focanotify","team_id":"T02CBDXTL4S","user_id":"U02CHLF0RK6","bot_id":"B02CBLC8N59","is_enterprise_install":false}
WARNING:werkzeug: * Debugger is active!
INFO:werkzeug: * Debugger PIN: 103-268-687
```
I can see auth.test  are succeeding. 

Note: I used ngrok proxy server.

**Comment by seratch** [comment]
@Ashok-senthilarasan When you use ngrok for serving public HTTPS URLs, there is no need to enable SSL on the Flask app side. The communications between your ngrok and your Flask app can be HTTP (not HTTPS) protocol and this is the common way to connect them:

>Slack end-user -(https)-> Slack server -(https)-> your ngrok endpoint -(http)-> your Flask app

So you can simply run http://127.0.0.1:5001. With that, forwarding ngrok requests to the http localhost URL should work for you.

If you still need some help on how to serve HTTPS endpoints on the Flask side, this is not a bolt-python specific question. You can ask for help somewhere else that you can get answers from Flask experts (such as stackoverflow, flask repo etc.)

I think that we've answered your question here in the best way possible. Would you mind closing this issue now?

**Comment by Ashok-senthilarasan** [comment]
Hi @seratch , Thanks for your thoughts.

I have been running Flask app for more than a year with self signed cert without issues. I used this to interact with block actions request from slack. didn't have problem. Now I am trying to add some new features to my app, in which the requests are taking more than 3 secs. I was facing issues that described [here](https://github.com/slackapi/python-slack-sdk/issues/788). I tried Python threading and Celery framework but it didn't help to fix it. so decided to use Bolt framwork. After I started use this framework I am running into this issue. Looks like it doesn't like self signed certs web server.   

Once again thanks for you help.
